integration-tests/braintree_test.go

```diff
@@ -31,30 +54,154 @@ func TestBraintree(t *testing.T) {
 	client := braintree.NewClient(&braintree.Credentials{
 		MerchantID: getEnv("BRAINTREE_MERCHANT_ID"),
 		PublicKey:  getEnv("BRAINTREE_PUBLIC_KEY"),
 		PrivateKey: getEnv("BRAINTREE_PRIVATE_KEY"),
 	})
 	authRequest := sleet_testing.BaseAuthorizationRequest()
-	authRequest.Amount = sleet.Amount{
-		Amount:   201000,
-		Currency: "USD",
+	auth, err := client.Authorize(authRequest)
+	if err != nil {
+		t.Errorf("Expected no error: received: %s", err)
 	}
-	authRequest.BillingAddress = &sleet.BillingAddress{
-		StreetAddress1: sPtr("77 Geary St"),
-		StreetAddress2: sPtr("Floor 4"),
-		Locality:       sPtr("San Francisco"),
-		RegionCode:     sPtr("CA"),
-		PostalCode:     sPtr("94108"),
-		CountryCode:    sPtr("US"),
+	if !auth.Success {
+		t.Errorf("Expected Success: received: %s", auth.ErrorCode)
 	}
-	resp, err := client.Authorize(authRequest)
+
+	captureRequest := &sleet.CaptureRequest{
+		Amount:               &authRequest.Amount,
+		TransactionReference: auth.TransactionReference,
+	}
+	capture, err := client.Capture(captureRequest)
+	if err != nil {
+		t.Error("Capture request should not have failed")
+	}
+
+	if capture.Success == false {
+		t.Error("Resulting capture should have been successful")
+	}
+}
+
+// TestBraintreeAuthPartialCapture
+//
+// Tests a Braintree authorization then a partial capture
+func TestBraintreeAuthPartialCapture(t *testing.T) {
+	client := braintree.NewClient(&braintree.Credentials{
+		MerchantID: getEnv("BRAINTREE_MERCHANT_ID"),
+		PublicKey:  getEnv("BRAINTREE_PUBLIC_KEY"),
+		PrivateKey: getEnv("BRAINTREE_PRIVATE_KEY"),
+	})
+	authRequest := sleet_testing.BaseAuthorizationRequest()
+	auth, err := client.Authorize(authRequest)
 	if err != nil {
 		t.Errorf("Expected no error: received: %s", err)
 	}
-	if resp.Success {
-		t.Errorf("Expected not-Success: received: %s", resp.ErrorCode)
+	if !auth.Success {
+		t.Errorf("Expected Success: received: %s", auth.ErrorCode)
+	}
+
+	captureRequest := &sleet.CaptureRequest{
+		Amount: &sleet.Amount{
+			Amount:   50,
+			Currency: "USD",
+		},
+		TransactionReference: auth.TransactionReference,
+	}
+	capture, err := client.Capture(captureRequest)
+	if err != nil {
+		t.Error("Capture request should not have failed")
+	}
+
+	if capture.Success == false {
+		t.Error("Resulting capture should have been successful")
+	}
+}
+
+// TestBraintreeAuthVoid
+//
+// This should successfully create an authorization on Braintree then Void/Cancel the Auth
+func TestBraintreeAuthVoid(t *testing.T) {
+	client := braintree.NewClient(&braintree.Credentials{
+		MerchantID: getEnv("BRAINTREE_MERCHANT_ID"),
+		PublicKey:  getEnv("BRAINTREE_PUBLIC_KEY"),
+		PrivateKey: getEnv("BRAINTREE_PRIVATE_KEY"),
+	})
+	authRequest := sleet_testing.BaseAuthorizationRequest()
+	auth, err := client.Authorize(authRequest)
+	if err != nil {
+		t.Error("Authorize request should not have failed")
+	}
+
+	if auth.Success == false {
+		t.Error("Resulting auth should have been successful")
+	}
```

Comment:
I fee like we don't need to have these checks everywhere since we already have a test which checks for valid auth for `sleet_testing.BaseAuthorizationRequest`

Reply:
Yeah I dont know I kind of looked around at other examples and I kind of like the pattern of continually building checks. Even if its redundant, if I don't have those checks and make assumptions that the auth was successful then what if capture fails down below and I don't know why?

Reply:
Ya makes sense. This pattern works!